ui/src/pages/gcp/login.js

```diff
@@ -0,0 +1,329 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React from 'react';
+import Button from '../../components/denali/Button';
+import API from '../../api';
+import Error from '../_error';
+import styled from '@emotion/styled';
+import createCache from '@emotion/cache';
+import RequestUtils from '../../components/utils/RequestUtils';
+import { selectIsLoading } from '../../redux/selectors/loading';
+import { selectUserResourceAccessList } from '../../redux/selectors/user';
+import { getUserResourceAccessList } from '../../redux/thunks/user';
+import { connect } from 'react-redux';
+import { CacheProvider } from '@emotion/react';
+import { ReduxPageLoader } from '../../components/denali/ReduxPageLoader';
+import {
+    getRolesFromResourceAccessList,
+    getProjectName,
+    getProjectRoleName,
+} from '../../server/utils/apiUtils';
+import {
+    USER_DOMAIN,
+    GENERIC_GCP_ERROR,
+} from '../../components/constants/constants';
+
+const GcpHeader = styled.header`
+    border-bottom: 1px solid lightgray;
+    margin: 10px;
+    margin-bottom: 20px;
+`;
+
+const ParentWrapperDiv = styled.div`
+    max-width: 800px;
+    margin: 0 auto;
+`;
+
+const StyledP = styled.p`
+    font-size: 16px;
+`;
+
+const RadioButtonsContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+    margin-top: 10px;
+    margin-bottom: 10px;
+    margin-left: 20px;
+    font-weight: bold;
+`;
+
+const ProjectTitleDiv = styled.div`
+    font-size: 18px;
+    display: inline;
+    border-bottom: 1px solid lightgray;
+    margin-bottom: 10px;
+    padding: 10px;
+    width: 100%;
+    display: block;
+`;
+
+const RadioButton = styled.input`
+    vertical-align: middle;
+    width: 18px;
+    height: 18px;
+    margin: 10px;
+`;
+
+const ProjectLabel = styled.label`
+    margin: 0;
+    font-weight: bold;
+    font-size: 16px;
+    color: #000;
+    vertical-align: middle;
+`;
+
+const SubmitContainer = styled.div`
+    display: flex;
+`;
+
+export async function getServerSideProps(context) {
+    const api = API(context.req);
+    let reload = false;
+    let notFound = false;
+    let error = null;
+    const domains = await Promise.all([api.getForm()]).catch((err) => {
+        let response = RequestUtils.errorCheckHelper(err);
+        reload = response.reload;
+        error = response.error;
+        return [{}];
+    });
+
+    let queryParams = context.query || {};
+    let isAdmin = queryParams.isAdmin === 'true';
+    let projectDomainName = queryParams.projectDomainName || '';
+    let serverSideError = queryParams.error || '';
```

Comment:
I don't think we should set if an error has occurred in query parameter. Once case to consider: 
If an occurred in server side (let's say an intermittent issue with zms), user will be redirected to URL with `error=true`, now the user will try to refresh the page and even though the issue may have been fixed on the server side, we will still show the error page since error param is set to true.

Reply:
updated validation error logic